website/docs/docs/building-a-dbt-project/building-models/python-models.md

```diff
@@ -0,0 +1,378 @@
+
+### Referencing other models
+
+Your Python model will want to read data from other models (SQL or Python). Do this using the `dbt.ref()` function. The same idea applies for raw source tables, via `dbt.source()`. Those functions return data frames, pointing to the upstream source, model, seed, or snapshot. 
+
+<File name='models/my_python_model.py'>
+
+```python
+def model(dbt, session):
+  
+    # data frame representing an upstream model
+    upstream_model = dbt.ref("upstream_model_name")
+    
+    # data frame representing an upstream source
+    upstream_source = dbt.source("upstream_source_name", "table_name")
+```
+
+</File>
+
+Of course, you can `ref()` your Python model in downstream SQL models, too:
+
+<File name='models/downstream_model.sql'>
+
+```sql
+with upstream_python_model as (
+  
+    select * from {{ ref('my_python_model') }}
+  
+),
+
+...
+```
+
+</File>
+
+### Configuring Python models
+
+Just like with SQL models, there are three ways to configure Python models:
+1. In `dbt_project.yml`, where you can configure many models at once
+2. In a dedicated `.yml` file, within the `models/` directory
+3. Within the model's `.py` file, using the `dbt.config()` function
+
+The `dbt.config()` function allows you to set configurations for your model:
+
+<File name='models/my_python_model.py'>
+
+```python
+def model(dbt, session):
+    
+    # setting configuration
+    dbt.config(materialized="table")
+```
+
+</File>
+
+The `config()` function accepts _only_ literal values (strings, booleans, and numeric types). It is not possible to pass another function or more complex data structure. The reason: dbt statically analyzes the arguments to `config()` while parsing your model, without actually executing any of your Python code.
+
+Python models have limited access to project context, including inputs such as `var`, `env_var`, and `target`. If you want to use those values to power conditional logic in your model, we recommend passing them through a dedicated `.yml` file config instead:
+
+<File name='models/config.yml'>
+
+```yml
+version: 2
+
+models:
+  - name: my_python_model
+    config:
+      materialized: table
+      target_name: "{{ target.name }}"
+      specific_env_var: "{{ env_var('SPECIFIC_ENV_VAR') }}"
+```
+
+</File>
+
+Then, use the `dbt.config.get()` function to _access_ values of configurations that have been set:
+
+<File name='models/my_python_model.py'>
+
+```python
+def model(dbt, session):
+    ...
+    target_name = dbt.config.get("target_name")
+    orders_df = dbt.ref("fct_orders")
+    
+    # limit data in dev
+    if target_name == "dev":
+        orders_df = orders_df.limit(500)
+```
+
+</File>
+
+### Materializations
+
+Python models support two materializations:
```

Comment:
@jtcohen6 is it worth adding a call out as to why it can't materialize as a view and ephemeral?

On a similar train of thought, can you configure these as snapshots? I assume not?

Reply:
Good point re: ephemeral, and clarifying no support for non-model resource types (snapshots, tests).

Views are tricky. I don't want to get _too_ into the weeds here. Views actually are supported by [Snowpark](https://docs.snowflake.com/en/developer-guide/snowpark/python/working-with-dataframes.html#creating-a-view-from-a-dataframe), but there isn't an analogue in PySpark. Up to us if we want to add it in a platform-specific way.